front-end/src/components/gallery/Gallery.js

```diff
@@ -3,16 +3,24 @@ import styled from 'styled-components';
 import Filter from "../filter/Filter";
 import Product from '../product/Product';
 import DisplayFilters from '../filter/DisplayFilters';
+import { Link, useHistory, useLocation } from 'react-router-dom';
 import { productService } from "../../services/productServices";
 import { useDispatch, useSelector } from "react-redux";
 import { selectFilters, selectProduct } from "../../redux/productSlice";
+import { utilFunc } from "../../services/utils";
+import { constants } from "../../constants/constants";
 
 const Container = styled.div`
     width: 80%;
     min-height: 100vh;
     background-color: whitesmoke;
     margin: 0 auto 0 auto;
```

Comment:
This can be simplified:
```suggestion
    margin: 0 auto;
```
When you have only two arguments first is for top and bottom and second is for right and left.